### ISSUE TITLE
Importance scores

### ISSUE THREAD

**Issue by rhagan94** [GET-Foundation/get_model#31]
Hi Xi,

I have successfully ran the finetune tutorial on my data and would like to extract importance scores for all of my peaks and motifs - essentially so that I can rank each peak and motif by importance and then link the most important to specific genes (similar to the enhancer-gene pair prediction in your paper)

Can you suggest the best approach for this please?

Any help is greatly appreciated!

Ryan

**Comment by fuxialexander** [comment]
Oops...sorry I was buried in calendars (and procrastination) in the past weeks...

Regarding the enhancer-gene pair prediction: I added [this](https://github.com/GET-Foundation/gcell/blob/883e55872854fd2b1491df4326a2909aa7067b84/src/gcell/cell/celltype.py#L1817) function, which should approximately give you what we have in the paper. The difference is that in the paper we trained a simple distance->Hi-C CNN model to produce a more accurate power law (input is only distance, output is HiC log count); in the function we used the powerlaw in ABC-model trained on K562. Note that there is probably many different ways to produce a score, but currently the benchmark data for this problem is still limited...

To use this function, you should train a `quantitative_atac` model, and make sure use the updated docker because I added accommodations in the `interpretation_step` to save gradient in the `region_embed` layer, which is required when we calculate the region importance score

**Comment by rhagan94** [comment]
Thanks Xi, that's very helpful! Would it be possible to provide the code for the CNN model that we could adapt and use to generate a more accurate powerlaw? - there is Hi-C data available for the tissue type I used to finetune the model

**Comment by fuxialexander** [comment]
Model is here: https://github.com/GET-Foundation/get_model/blob/c25885c8899d992a24cc87921b39993bcf0760c5/get_model/model/model.py#L749

The dataset code is a bit too complex to accommodate into the current branch. But since this is a distance-based model, you can just write a customized dataloader: Get your ATAC peaks in a bed/csv file, use this function https://github.com/GET-Foundation/get_model/blob/c25885c8899d992a24cc87921b39993bcf0760c5/get_model/dataset/zarr_dataset.py#L70 to get their Hi-C log count, and calculate their pairwise relative 1D distance (which is a 2D matrix). The input is this pairwise relative distance matrix, and output would be your Hi-C log count.

**Comment by rhagan94** [comment]
Thank you! I'm just getting round to looking at this now and have a few questions. Does it require peaks on the same chromosome and within 4Mb of each other as input? - generating a list of 2D pairwise matrices that encapsulate all the peaks which are then input to the model?

Thanks again for your help

**Comment by fuxialexander** [comment]
> Does it require peaks on the same chromosome and within 4Mb of each other as input?
Yes, that's for one input sample. You can chunk the chromosomes into 2-4Mb bins. Usually the default 200 peaks won't exceed that.
